staking/programs/integrity-pool/src/state/pool.rs

```diff
@@ -347,6 +360,33 @@ impl PoolData {
         Ok(())
     }
 
+    /**
+     * Adjusts the rewards to be distributed if the pool reward custody balance is not enough to
+     * cover the rewards
+     */
+    pub fn adjust_rewards_if_needed(
+        &self,
+        y: u64,
+        total_eligible_delegation: u64,
+        pool_reward_custody_balance: u64,
+    ) -> Result<(u64, u64)> {
+        let rewards_to_be_distributed: u64 =
+            ((u128::from(total_eligible_delegation) * u128::from(y)) / FRAC_64_MULTIPLIER_U128)
+                .try_into()?;
+        if rewards_to_be_distributed + self.claimable_rewards > pool_reward_custody_balance {
+            let adjusted_y = ((u128::from(y)
+                * u128::from(pool_reward_custody_balance - self.claimable_rewards))
+                / u128::from(rewards_to_be_distributed))
+            .try_into()?;
+            let adjusted_rewards_to_be_distributed =
+                ((u128::from(rewards_to_be_distributed) * u128::from(adjusted_y)) / u128::from(y))
+                    .try_into()?;
+            Ok((adjusted_y, adjusted_rewards_to_be_distributed))
+        } else {
+            Ok((y, rewards_to_be_distributed))
+        }
```

Comment:
i would write it this way
```
let adjusted_reward_to_be_distributed = min(reward_to_be_distributed, pool_reward_custody_balance - self.claimable_rewards);
let adjusted_y = y * adjust_reward_to_be_distributed / reward_to_be_distributed
```

Reply:
I updated it, but I think your way has bigger rounding errors.